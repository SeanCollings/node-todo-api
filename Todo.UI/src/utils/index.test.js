import { createRandomId, decodeJWT, sortByCreatedDate } from '.';
import jwtDecode from 'jwt-decode';

jest.mock('jwt-decode');

describe('utils/index - unit test', () => {
  test('createRandomId', () => {
    const id = createRandomId();
    expect(id).toHaveLength(5);
  });

  test('sortByCreatedDate', () => {
    const unsortedArray = [
      {
        createdAt: '2021-03-03T19:16:08.047Z',
      },
      {
        createdAt: '2021-02-03T19:16:08.047Z',
      },
      {
        createdAt: '2021-01-03T19:16:08.047Z',
      },
      {
        createdAt: '2021-03-01T19:16:08.047Z',
      },
      {
        createdAt: '2020-03-03T19:16:08.047Z',
      },
    ];
    const sortedArray = sortByCreatedDate(unsortedArray);
    expect(sortedArray).toMatchSnapshot();
  });

  describe('decodeJWT', () => {
    test('should return null if no token is passed in', () => {
      const decoded = decodeJWT();
      expect(decoded).toBeNull();
    });

    test('should return part of the decoded jwt', () => {
      jwtDecode.mockReturnValueOnce({ name: 'name' });
      const decoded = decodeJWT('jwt');
      expect(decoded).toEqual({ name: 'name' });
    });
  });
});
