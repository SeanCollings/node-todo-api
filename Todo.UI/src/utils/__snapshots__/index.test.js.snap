// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/index - unit test sortByCreatedDate 1`] = `
Array [
  Object {
    "createdAt": "2020-03-03T19:16:08.047Z",
  },
  Object {
    "createdAt": "2021-01-03T19:16:08.047Z",
  },
  Object {
    "createdAt": "2021-02-03T19:16:08.047Z",
  },
  Object {
    "createdAt": "2021-03-01T19:16:08.047Z",
  },
  Object {
    "createdAt": "2021-03-03T19:16:08.047Z",
  },
]
`;
